src/blueprint/operators/grpc.rs

```diff
@@ -131,16 +131,22 @@ impl TryFrom<String> for GrpcMethod {
     fn try_from(value: String) -> Result<Self, Self::Error> {
         let method: Vec<&str> = value.split('.').collect();
 
-        if method.len() != 3 {
+        if method.len() < 3 {
             return Err(ValidationError::new(format!(
-                "Invalid method format: {}. Expected format is <package/proto_id>.<service>.<method>",
+                "Invalid method format: {}. Expected format is <link_id>.<package.service>.<method>",
                 value
             )));
         }
 
         let id = method[0].to_string();
-        let service = format!("{}.{}", id, method[1]);
-        let name = method[2].to_string();
+        // TODO: some of implementation (like tonic) will use file name as default package name
+        // if the package is not specified in proto explicitly, that affect the request url.
+        // but `prost_reflect` crate won't find the service if we try to specify the package name if there is
+        // no package definition in proto file, so we may end up with case that we forced to specify
+        // package name to be able to get response, but in the same time we can't do that
+        // because prost won't find that definition for same proto file.
+        let service = method[1..method.len() - 1].join(".");
+        let name = method.last().unwrap().to_string();
```

Comment:
make this safe. Handle errors

Reply:
we have check for `method.len() < 3` earlier, this is not necessary

Reply:
I think we should get change the purpose of link id. It should give user ability to define a name in case there is no file name or package provided. We should by default use package id everywhere.

something like https://github.com/tailcallhq/tailcall/blob/b543c748ea793696da0552e8627b48d05008b2c3/src/config/reader.rs#L119 

check #1227

Reply:
> but this will force users to define methods like `news.news.NewsService.GetAllNews` (from modified grpc.graphql example) which is incorrect if package id is `news`. Method should be defined as `news.NewsService.GetAllNews`

it'll work here since the first news is the name of the link, and only `news.NewsService.GetAllNews` will be used as service name. 
But that looks awkward actually, let's discuss your suggestion, I left some comments there